format/FlightSql.proto

```diff
@@ -1475,6 +1549,35 @@ message CommandStatementQuery {
 
   // The SQL syntax.
   string query = 1;
+  // Include the query as part of this transaction (by default queries are auto-committed).
```

Comment:
from an RPC optimization standpoint, it seems that maybe we want a way of specifying this should be considered the first action inside of a transaction?

Reply:
It seems like maybe there should be something like: 

```
message TransactionDetails {
   oneof transaction {
     ActionBeginTranscation begin_transaction = 1;
     bytes existing_transaction_id = 2;
   }
}
```
that can be included.

Reply:
The intent was to mimic prepared statements: the server assigns the transaction ID and gives it to the client. So here there's not a great way to return the transaction ID to the client. It would be good if FlightInfo could gain an app_metadata field for such things (since Tickets are not meant to be client-introspected).

Reply:
I suppose we could use schema metadata for that, though.

Reply:
it seems app_metadata would probably be generally useful.  It seems like a hack to force the info onto the schema.

Reply:
I'll propose that separately when I get a chance (and I'll see if I can set aside some time to help with the small-result optimization stuff)